### DISCUSSION TITLE
Any recommendation for data fetching within dialogs?

### DISCUSSION THREAD

**Discussion by rossfishkind** [remix-run/remix#4657] [Q&A] [answered]
Hi all! I've been loving working with Remix, but I'm having a hard time figuring out the optimal approach to data fetching within dialogs.

Here's an example:

> I have an app where customers can create proposals. A customer clicks on one of their proposals and it brings them to a new page to edit it (loading data via the `loader` function). From there, they are interested in editing the associated client for that proposal, so they click on that client, and I open a dialog showing the relevant form fields to edit that client. In addition to those fields, there is also a dropdown with a list of all of their clients so that they can easily swap out the client for that proposal.
> 
> That client list needs to be populated from somewhere. Should I fetch it on the client-side? Should I fetch it via the `loader` function when the underlying page loads? Should I make the modal a sub-route of the underlying page so that I can handle data fetching via it's own `loader` as well as an `action` down the line?

I'm curious what others have done in this scenario and if there is a recommended approach in Remix. Thanks!

**Comment by machour** [comment]
Not sure if it fits your use case, but there's a route modal example here: https://github.com/remix-run/examples/tree/main/route-modal

Full stack components may also answer your use case:  https://www.epicweb.dev/full-stack-components

**Comment by rossfishkind** [comment, marked as answer]
@machour ah yes, thank you! Those were helpful. What I took away from that is Kent's use of `useFetcher` with resource routes, which the Remix documentation says is helpful to `fetch data not associated with UI routes (popovers, dynamic forms, etc.)`. That makes complete sense, just something I must've missed. Thanks again!
